### PR TITLE
Code Example Toggle Fix

### PR DESCRIPTION
**Pull request by bh247484**
#### What does this PR do?

Fixes broken Code Example toggle.  Subdivides `site.js` into proper area of concerns and imports JS only where it is needed (`samples.html.erb` vs `application.html.erb` layout).

#### Breaking Changes

No

#### Runway Ticket URL

n/a bug hotfix

### CODE REVIEW

File: app/views/layouts/playbook/application.html.erb [hunk 1]
```diff
@@ -23,8 +23,8 @@
   </body>
   <%= javascript_include_tag "https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.js" %>
   <% if @type == 'react' %>
-    <%= javascript_packs_with_chunks_tag 'application', 'react-examples' %>
+    <%= javascript_packs_with_chunks_tag 'application', 'react-examples', 'site' %>
   <% else %>
-    <%= javascript_packs_with_chunks_tag 'application', 'rails-examples' %>
+    <%= javascript_packs_with_chunks_tag 'application', 'react-examples', 'site' %>
```
Comment (coleerikson) [inline app/views/layouts/playbook/application.html.erb:28]: Is this correct?

Comment (bh247484) [reply app/views/layouts/playbook/application.html.erb:28]: Good catch!  I think I copy and pasted.

Comment (coleerikson) [reply app/views/layouts/playbook/application.html.erb:28]: Should be `rails` no?

File: app/views/layouts/playbook/samples.html.erb [hunk 2]
```diff
@@ -16,9 +16,9 @@
...
 
     <% if @type == 'react' %>
-      <%= javascript_packs_with_chunks_tag 'application', 'react-examples', 'samples', 'site' %>
+      <%= javascript_packs_with_chunks_tag 'application', 'react-examples', 'samples', 'full-page-samples' %>
     <% else %>
-      <%= javascript_packs_with_chunks_tag 'application', 'rails-examples', 'samples', 'site' %>
+      <%= javascript_packs_with_chunks_tag 'application', 'rails-examples', 'samples', 'full-page-samples' %>
       <%= yield :pb_js %>
     <% end %>
 </html>
```
Comment (coleerikson) [inline app/views/layouts/playbook/samples.html.erb:24]: Add a space if you need to commit anything else.

**Review by coleerikson — APPROVED** [review]
LGTM besides my comments.

File: app/views/layouts/playbook/samples.html.erb (hunk 2 above)
Comment (coleerikson) [reply app/views/layouts/playbook/samples.html.erb:24]: a line that it.

**Review by jasperfurniss — APPROVED** [review]

**Pull request merged**